### DISCUSSION TITLE
When creating a translation, it defaults to English:US instead of the default language in the project settings

### DISCUSSION THREAD

**Discussion by sxsws** [directus/directus#13412] [Feature Requests]
When the project settings default language is set to something other than EN:US, the first translation when making a new translation still defaults to EN:US instead of whatever the project default is.

**Comment by azrikahar** [comment]
Technically project settings default language was added quite recently via #8196, so let's treat this as a feature request. Just to be sure, you are mentioning the translations interface within an item right? Though we may also need to propagate this elsewhere like collection/field translations or translated strings as well to be really thorough.

**Reply by sxsws** [reply]
Places I've seen this happen so far: field translation, folder translation (folder in data model) so I'd assume it's all if not most places it's possible to add a translation.

**Comment by miagg** [comment]
I have the same problem. I'm trying to do the following but without success.
1. Make the translations preview display the default language translation
2. When creating a new translation, choose the default language instead of the first language in the db

Isn't this highlighted option supposed to do all of the above? For me it does not work.
![img](https://b2.apptime.gr/file/apptime/Screenshot-2022-05-26-at-10.04.24-AM.png)

**Reply by azrikahar** [reply]
> Make the translations preview display the default language translation

It should work, but what is the primary key values of your language table? Unless you're not using the autogenerated language table, it should be `en-US`, `fr-FR` etc as Default Language instead of a numeric ID.

> When creating a new translation, choose the default language instead of the first language in the db

When creating a new translation, Display related configurations doesn't really apply as we are using interface, not a display.

However I do agree that it would be a nice improvement as an option in Interface (currently there's only Language Indicator Field option)

**Reply by miagg** [reply]
> It should work, but what is the primary key values of your language table? Unless you're not using the autogenerated language table, it should be en-US, fr-FR etc as Default Language instead of a numeric ID.

I'm using a custom languages table with numeric ids as PK. I added the alias field by manually selecting PKs and intermediate table. Everything works except for the default language. I guess I'm out of luck. :(

**Reply by azrikahar** [reply]
Hmm I'm not quite sure I can visualize the setup you are describing here, would you mind sharing a quick screenshot of a particular language item (and maybe show the primary key if it's hidden)?

**Reply by miagg** [reply]
![img](https://b2.apptime.gr/file/apptime/Screenshot-2022-05-26-at-11.50.10-AM.png)

I'm using Postgres by the way.

**Reply by azrikahar** [reply]
Would you mind sharing a schema snapshot so that I can attempt to investigate it better? Thanks!

**Reply by miagg** [reply]
sure! Thanks for your support.
[schema.yaml.zip](https://github.com/directus/directus/files/8785121/schema.yaml.zip)

**Reply by azrikahar** [reply]
There is this line, which I figured should've made it work when you enter default language as `2` (especially the `val?.[relation.value.field]?.[langPkField] === props.defaultLanguage` condition check):

https://github.com/directus/directus/blob/06cd1186cf955cdde14f1cfe02375b96e7e88b73/app/src/displays/translations/translations.vue#L96-L98

However you are using _integer_ primary key, so in terms of strict equality, it fails, as seen here:

![chrome_JpgDgwam5j](https://user-images.githubusercontent.com/42867097/171619197-4b3039ed-0185-48e2-be4c-2646df038dc0.png)

and here's a brief testing on changing it to loose equality instead, and it does work now:

https://user-images.githubusercontent.com/42867097/171619216-d7f77e69-489b-4d1e-b7c2-d6f2e5db2426.mp4

@rijkvanzanten should we just cast `val?.[relation.value.field]?.[langPkField]` to string here? There shouldn't be any problem with such change right? Can't think of any off the top of my head 🤔

**Reply by miagg** [reply]
This is exactly what I thought the problem might be. I was expecting the value to cast to an integer since it's numeric so that strict equality would work by trying entering a raw value of 2 but I didn't have any luck. I can't think of a reason loose equality would cause any harm here also. Thanks for tracking this!

**Reply by rijkvanzanten** [reply]
> @rijkvanzanten should we just cast val?.[relation.value.field]?.[langPkField] to string here? There shouldn't be any problem with such change right? Can't think of any off the top of my head 🤔

Works for me 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
